src/makeCalculator.js

```diff
@@ -4,7 +4,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(num) {
+      this.result -= num;
+    },
+
+    divide(num) {
+      this.result /= num;
+    },
+
+    multiply(num) {
+      this.result *= num;
+    },
+
+    operate(operation, num) {
+      operation.call(this, num);
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
```

Comment:
The `reset` method correctly resets the `result` to 0 and returns `this` to enable method chaining.